File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        lines = matrix_string.splitlines()
        self.rows = [[int(s) for s in line.split()]
                     for line in lines]
        self.columns = [[row[j] for row in self.rows]
                        for j in range(len(self.rows[0]))]

    def row(self, row_number):
        return self.rows[row_number - 1]

    def column(self, column_number):
        return self.columns[column_number - 1]

File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(matrix_string) {
    const lines = matrix_string.split("\n");
    this._rows = lines.map(line => line.split(" ").map(s => Number(s)));
    this._columns = Array.from({ length: this._rows[0].length }, (x, i) =>
      this._rows.map(row => row[i])
    );
  }

  get rows() {
    return this._rows;
  }

  get columns() {
    return this._columns;
  }
}
